EIPS/eip-4844.md

```diff
@@ -234,6 +232,41 @@ def get_origin(tx: SignedBlobTransaction) -> Address:
     return ecrecover(tx_hash(tx), int(sig.y_parity)+27, sig.r, sig.s)
 ```
 
+### Header extension
+
+The current header encoding is extended with a new 256-bit unsigned integer field `blob_txs`. This is the running
+total of excess blob txs included on chain since this EIP was activated. The resulting RLP encoding of the header is therefore:
+
+```
+rlp([
+    parent_hash,
+    ommers_hash,
+    coinbase,
+    state_root,
+    tx_root,
+    receipt_root,
+    bloom,
+    difficulty,
+    number,
+    gas_limit,
+    gas_used,
+    time,
+    extra,
+    mix_digest,
+    nonce,
+    base_fee,
+    blob_txs
+])
+```
+
+The value of `blob_txs` can be calculated using the parent header and number of blob transactions in the block.
+
+```python
+def calc_excess_blob_txs(parent: Header, blobs: int) -> int:
+    adjusted = parent.blob_txs + blobs
+    return adjusted - min(TARGET_BLOBS_PER_BLOCK, adjusted)
```

Comment:
This changes the mechanism, because it only stores the diff if positive. To keep the mechanism intact, `blob_txs` would have to be a signed integer. If this is impractical, I would recommend sticking to the original `actual_total` accounting variant.

Reply:
So my understanding of the current mechanism is that the `max` function clips any negative deltas vs. the target amount and thus causes the same behavior:

```python
def update_blob_gas(state: ExecState, blob_txs_in_block: int):
    current_total = read_storage(
        state,
        SYSTEM_STATE_ADDRESS,
        TOTAL_BLOB_TXS_STORAGE_SLOT
    )
    # Don't let the new total fall behind the targeted total to avoid
    # accumulating a long period of very low fees
    blocks_since_start = get_current_block(pre_state) - FORK_BLKNUM
    targeted_total = blocks_since_start * TARGET_BLOB_TXS_PER_BLOCK
    new_total = max(current_total + blob_txs_in_block, targeted_total)
    write_storage(
        state,
        SYSTEM_STATE_ADDRESS,
        TOTAL_BLOB_TXS_STORAGE_SLOT,
        new_total
    )
```

Reply:
Ah, you are correct, I overlooked that in the original EIP. Those two mechanisms are equivalent after all then.